### PR TITLE
Fixes for #149 and #148

### PR DESCRIPTION
**Pull request by avahowell**

### CODE REVIEW

File: js/mainjs/trayMenu.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	// Template for Sia-UI tray menu.
+	var menutemplate = [
+		{
+			label: 'Show Sia',
+			click: function () { window.show(); }
+		},
+		{ type: 'separator' },
+		{
+			label: 'Hide Sia',
+			click: function () { window.hide(); }
```
Comment (lukechampine) [inline js/mainjs/trayMenu.js:16]: could these just be `window.show` and `window.hide`?
although, even if it's possible, I'm not sure it's best practice.

**Comment by VoidingWarranties** [comment]
Can you close the issues in the commit messages both so that they are closed automatically and so that it's obvious what each commit fixes

File: js/mainjs/initWindow.js [hunk 2]
```diff
@@ -49,13 +48,12 @@ module.exports = function(config) {
...
 	// Load the index.html of the app.
 	mainWindow.loadURL('file://' + __dirname + '/../../index.html');
-
 	// Choose not to show the menubar
 	if (process.platform !== 'darwin') {
 		mainWindow.setMenuBarVisibility(false);
 	} else {
 		// Create the Application's main menu - OSX version might feel weird without a menubar
-		Menu.setApplicationMenu(AppMenu);
+		Menu.setApplicationMenu(new AppMenu(mainWindow));
```
Comment (lukechampine) [inline js/mainjs/initWindow.js]: I still haven't fully grasped the `new` operator. What purpose does it serve here?

File: index.js [hunk 3]
```diff
@@ -22,13 +23,14 @@ var config = require('./js/mainjs/config.js')(Path.join(__dirname, 'config.json'
 
 // When Electron loading has finished, start the daemon then the UI
 App.on('ready', function() {
+	// Load mainWindow
+	mainWindow = require('./js/mainjs/initWindow.js')(config);
 	// Load tray icon
-	var iconPath = Path.join(__dirname, 'assets', 'icon.png');
+	// OSX has a maximum tray icon height of 18px.
```
Comment (lukechampine) [inline index.js]: can probably drop this comment. It's more confusing than helpful.

Comment (VoidingWarranties) [inline index.js]: Where does 18px come from? http://iconhandbook.co.uk/reference/chart/osx/ says the maximum size is 22px. Also is it possible to use the previous larger icon (or vector image if we have that), but display it at a smaller size?

**Comment by VoidingWarranties** [comment]
Where does 18px come from for the OS X menu bar size? http://iconhandbook.co.uk/reference/chart/osx/ says the maximum size is 22px. Also is it possible to use the previous larger icon (or vector image if we have that), but display it at a smaller size?

File: js/mainjs/initWindow.js (hunk 2 above)
Comment (avahowell) [inline js/mainjs/initWindow.js]: I amended the commits to reflect their issue fixes.  I initially called this as Menu.setApplicationMenu(AppMenu()), and JSHint yelled at me for not calling a constructor using new.  Looking into it more, it looks like jshint expects all functions that start with caps to be constructors, invoked using new.  AppMenu() can still be called as a function, but should probably be renamed to reflect this convention

Comment (lukechampine) [inline js/mainjs/initWindow.js]: gotcha. I agree that `AppMenu` should be lowercased.

**Comment by avahowell** [comment]
Electron's nativeImage class only supports png and jpeg, unfortunately no vector images.  I chose 18px to give some vertical margin to the icon, the maximum tray icon size is indeed 22px on osx.

**Pull request merged**